scarb/src/flock.rs

```diff
@@ -37,6 +37,10 @@ impl FileLockGuard {
         self.path.as_path()
     }
 
+    pub fn parent(&self) -> &Utf8Path {
+        self.path.parent().unwrap()
```

Comment:
This was the way it's done in Cargo: 
https://github.com/rust-lang/cargo/blob/3c4c0a2b3c9df8f2e97b58358ce1b36ecc81bc5c/src/cargo/util/flock.rs#L56

And since we already copy the unpacking from Cargo, didn't see a reason to deviate in is particular case

Reply:
I'm fine with making unwrap more explicit as per offline discussion, my only concern is us handling similar operations in a slightly different way, e.g. just below we have: 
```rs
impl Deref for FileLockGuard {
    type Target = File;

    fn deref(&self) -> &Self::Target {
        self.file.as_ref().unwrap()
    }
}
```

Reply:
But it unwraps two totally different things, right? 🤔

Reply:
> But it unwraps two totally different things, right? 🤔

Sure, but it seems *a bit* strange to me that `file`, which is also optional, is unwrapped, while file's parent isn't